test/arg_max_min_test.js

```diff
@@ -0,0 +1,620 @@
+'use strict';
+
+import {argMax, argMin} from '../src/arg_max_min.js';
+import {Tensor} from '../src/lib/tensor.js';
+import * as utils from './utils.js';
+
+describe('test argMax and argMin', function() {
+  function testArgMaxMin(input, expected, func, options = {}) {
+    const inputTensor = new Tensor(input.shape, input.value);
+    const outputTensor = func(inputTensor, options);
+    utils.checkShape(outputTensor, expected.shape);
+    utils.checkValue(outputTensor, expected.value);
+  }
+
+  it('argMax default', function() {
+    testArgMaxMin(
+        {
+          shape: [3, 3],
+          value: [
+            1, 2, 3,
+            3, 0, 4,
+            2, 5, 2,
+          ],
+        },
+        {
+          shape: [],
+          value: [7],
+        },
+        argMax);
+  });
+
+  it('argMax axes=[0]', function() {
+    testArgMaxMin(
+        {
+          shape: [3, 3],
+          value: [
+            1, 2, 3,
+            3, 0, 4,
+            2, 5, 2,
+          ],
+        },
+        {
+          shape: [3],
+          value: [1, 2, 1],
+        },
+        argMax,
+        {
+          axes: [0],
+        });
+  });
+
+  it('argMax axes=[1]', function() {
+    testArgMaxMin(
+        {
+          shape: [3, 3],
+          value: [
+            1, 2, 3,
+            3, 0, 4,
+            2, 5, 2,
+          ],
+        },
+        {
+          shape: [3],
+          value: [2, 2, 1],
+        },
+        argMax,
+        {
+          axes: [1],
+        });
+  });
+
+  it('argMax 2D of shape [1, 3] axes=[1]', function() {
+    testArgMaxMin(
+        {
+          shape: [1, 3],
+          value: [1, 2, 3],
+        },
+        {
+          shape: [1],
+          value: [2],
+        },
+        argMax,
+        {
+          axes: [1],
+        });
+  });
+
+  it('argMax axes=[0, 1]', function() {
+    testArgMaxMin(
+        {
+          shape: [3, 3],
+          value: [
+            1, 2, 3,
+            3, 0, 4,
+            2, 5, 2,
+          ],
+        },
+        {
+          shape: [],
+          value: [7],
+        },
+        argMax,
+        {
+          axes: [0, 1],
+        });
+  });
+
+  it('argMax axes=[1, 0]', function() {
+    testArgMaxMin(
+        {
+          shape: [3, 3],
+          value: [
+            1, 2, 3,
+            3, 0, 4,
+            2, 5, 2,
+          ],
+        },
+        {
+          shape: [],
+          value: [7],
+        },
+        argMax,
+        {
+          axes: [1, 0],
```

Comment:
👍 Good that you have an axis-reversed case too in the argMinMax tests.

Reply:
+1